pkg/api/dashboard_permission.go

```diff
@@ -123,6 +137,64 @@ func (hs *HTTPServer) UpdateDashboardPermissions(c *models.ReqContext) response.
 	return response.Success("Dashboard permissions updated")
 }
 
+// updateDashboardAccessControl is used for api backward compatibility
+func (hs *HTTPServer) updateDashboardAccessControl(ctx context.Context, orgID, dashID int64, isFolder bool, items []*models.DashboardAcl, old []*models.DashboardAclInfoDTO) error {
+	commands := []accesscontrol.SetResourcePermissionCommand{}
+	for _, item := range items {
+		permissions := item.Permission.String()
+		role := ""
+		if item.Role != nil {
+			role = string(*item.Role)
+		}
+
+		commands = append(commands, accesscontrol.SetResourcePermissionCommand{
+			UserID:      item.UserID,
+			TeamID:      item.TeamID,
+			BuiltinRole: role,
+			Permission:  permissions,
+		})
+	}
+
+	for _, o := range old {
```

Comment:
random comment: the way in which we remove all the permissions and then add the new ones seems pretty convoluted to me. But I know that this logic was introduced before your changes, so I'm happy to leave it as it is, just wanted to complain 🙈

Reply:
Yes i agree but i don't want to change the API. The good thing is that this is all happening within one transaction